packages/firebase_ui_storage/lib/src/widgets/progress_indicator.dart

```diff
@@ -0,0 +1,176 @@
+import 'package:firebase_storage/firebase_storage.dart';
+import 'package:firebase_ui_shared/firebase_ui_shared.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+/// A builder that is invoked for each progress event.
+typedef TaskProgressBuilder = Widget Function(
+  BuildContext context,
+  double progress,
+);
+
+/// A builder that is invoked when an error occurs.
+typedef ErrorBuilder = Widget Function(
+  BuildContext context,
+  Object? error,
+);
+
+Widget _bindTask({
+  required Task task,
+  ErrorBuilder? errorBuilder,
+  required TaskProgressBuilder builder,
+}) {
+  return StreamBuilder<TaskSnapshot>(
+    stream: task.snapshotEvents,
+    builder: (context, snapshot) {
+      if (snapshot.hasError) {
+        return errorBuilder?.call(context, snapshot.error) ?? const SizedBox();
+      }
+
+      double progress;
+
+      if (!snapshot.hasData) {
+        progress = 0;
+      } else {
+        final taskSnapshot = snapshot.requireData;
+        progress = taskSnapshot.bytesTransferred / taskSnapshot.totalBytes;
+      }
+
+      return builder(context, progress);
+    },
+  );
+}
```

Comment:
Could this be a Widget class to benefit from hot reload?

Reply:
good catch, done